.github/workflows-source/ci-main.yml

```diff
@@ -154,11 +152,6 @@ jobs:
           BAZEL_COMMAND: "build"
           BAZEL_TARGETS: "//rs/..."
           BAZEL_CI_CONFIG: "--config=check --config=ci --keep_going"
-          # run on diff only if it is a pull request, otherwise run all targets
-          #RUN_ON_DIFF_ONLY: ${{ contains(github.event_name, 'pull_request') && 'true' || 'false'}}
-          # TODO: disabling until the following issue is resolved
-          # https://github.com/dfinity/ic/actions/runs/9699415138/job/26768332602
-          RUN_ON_DIFF_ONLY: false
```

Comment:
We had this logic disabled here. Your changes don't allow disabling this logic on the CI job level - or am I seeing that wrong? So what do we do with this unresolved issues and any potentially similar issue that might arise?

Reply:
Ah okay good point - yeah let me change some of that back.

Reply:
We could just check where we invoke `build-diff.sh` if there's env variable set on the job and have all targets (`//...`), specified if that's the case. Or not?